resources/qml/Dialogs/DiscardOrKeepProfileChangesDialog.qml

```diff
@@ -98,9 +100,26 @@ UM.Dialog
 
     buttonSpacing: UM.Theme.getSize("thin_margin").width
 
-    leftButtons: [
+    Rectangle
+    {
+        // Use a rectangle to get access to states. For some reason top-levels like Dialog/Window ect. don't have them.
+        // NOTE: The default state is 'switch profiles', alternate states are used for 'save from [built-in|custom]'.
+        id: alternateStates
```

Comment:
I think you want to use a boolean or enum with a property for this. 

States are normally used for changing the UI automagically by evaluating the "when" expression. Not for storing the state and pulling it back out elsewhere. The state name is intended for use in Animations an Behaviours.

Reply:
I was a bit more broad in my head originally. I didn't go back and rescope. Will alter this.

Reply:
Should be fixed now :-)